scripts.py

```diff
@@ -651,6 +664,258 @@ def get_category_fieldnames(self, rows, existing):
+                    'different CustomList name to create a new list.'
+                )
+            return None
+
+        if custom_list.data_source != self.source:
+            raise self.UneditableCustomList(
+                '%r was generated by an internal process and cannot be '
+                'edited by puny humans.'
+            )
+
+        return custom_list
+
+    def works_from_source(self, source_csv):
+        """Extracts works from the source file and does any work
+        requested by any `SELECTION_HEADERS`.
+
+        :return: A query of Works best representing URNs from the CSV.
+        """
+        filename = os.path.abspath(source_csv)
+
+        identifiers = list()
+        selections = defaultdict(list)
+        # initialize selector lists in the dict.
+        [selections[selector] for selector in self.SELECTION_HEADERS]
+
+        with open(filename) as f:
+            reader = csv.DictReader(f)
+            for row in reader:
+                urn = row.get('urn')
+                identifier = Identifier.parse_urn(self._db, urn)[0]
+                identifiers.append(identifier)
+
+                for selector in self.SELECTION_HEADERS:
+                    if row.get(selector):
+                        selections[selector].append(identifier)
+
+        works_qu = Work.from_identifiers(self._db, identifiers)
+        works_qu = works_qu.options(
+            joinedload(Work.license_pools),
+            contains_eager(Work.presentation_edition)
+        )
+
+        youth_works_qu = None
+        selections_for_youth = selections['youth']
+        if selections_for_youth:
+            youth_works_qu = Work.from_identifiers(
+                self._db, selections_for_youth, base_query=works_qu
+            )
+
+        # Make sure we have the most up-to-date versions of the work,
+        # according to whatever selections have been made.
+        self._update_works(selections)
+
+        return works_qu, youth_works_qu
+
+    def _update_works(self, selections_dict):
+        """Updates works according to a set of human-generated selections."""
+
+        rejected_covers = selections_dict['hide_cover']
+        if rejected_covers:
+            Work.reject_covers(self._db, rejected_covers)
+
+        featured_work_identifiers = selections_dict['featured']
+        for identifier in featured_work_identifiers:
+            work = identifier.licensed_through.work
+            featured_quality = Configuration.minimum_featured_quality()
+            if not work.quality > featured_quality:
+                # This creates a single measurement attributed to
+                # this DataSource and normalizes its value to equal the
+                # minimum requirements for being featured.
+                measurement = identifier.add_measurement(
+                    self.source, Measurement.RATING, 1
+                )
+                measurement._normalized_value = featured_quality
```

Comment:
I agree that changing _normalized value is icky. One way of fixing it would be to add a `Work.featured` field which could be set independently of Work.quality. That would require some significant changes in core though.

A simpler way would be to change this `add_measurement` call, which I think you should do anyway. If someone looked at the book and explicitly said "feature this sucker!" then that's a very heavy vote (`weight=1000` or something), but it's not a vote for `quality=1.0`. It's a vote for the minimum quality necessary to feature a book. (`Configuration.MINIMUM_FEATURED_QUALITY`, which is usually 0.65, I think).

Since you're only making this intervention if the existing work quality was below the minimum feature quality, this change will do more or less what you want. It will put a book that previously had a bad score up to the minimum necessary to be featured. A book that was already featured because of its super-good score will be left alone, and its super-good score won't be dragged _down_ to the minimum necessary score.